efs-lib/src/cmr/efs/connection.clj

```diff
@@ -26,19 +46,27 @@
 (defn save-concept
   "Saves a concept to EFS"
   [provider concept-type concept]
-  (let [concept-path (format "%s/%s/%s/%s.r%d.zip" (efs-config/efs-directory) (:provider-id provider) (name concept-type) (:concept_id concept) (:revision_id concept))]
+  (let [concept-path (format "%s/%s/%s/%s/%s.r%d.zip" (efs-config/efs-directory) (:provider-id provider) (name concept-type) (:concept_id concept) (:concept_id concept) (:revision_id concept))]
     (info "Saving concept to EFS at path " concept-path)
     (io/make-parents (io/file concept-path))
     (Files/write (Paths/get concept-path (into-array String [])) (:metadata concept) (into-array OpenOption []))))
 
 (defn get-concept
   "Gets a concept from EFS"
-  ([provider concept-type concept-id])
-  ([provider concept-type concept-id revision-id]))
+  ([provider concept-type concept-id]
+   (get-concept provider concept-type concept-id (get-latest-revision (:provider-id provider) (name concept-type) concept-id)))
+  ([provider concept-type concept-id revision-id]
+   (when revision-id
+     (let [concept-path (format "%s/%s/%s/%s/%s.r%d.zip" (efs-config/efs-directory) (:provider-id provider) (name concept-type) concept-id concept-id revision-id)]
+       (try
+         (info "Getting concept from EFS at path " concept-path)
+         {:revision-id revision-id :metadata (util/gzip-bytes->string (Files/readAllBytes (Paths/get concept-path (into-array String [])))) :deleted false}
+         (catch Exception _e nil))))))
```

Comment:
I think I'll go ahead and log it for now. My expectation right now is that most errors would be a result of the revision not existing and causing a file doesn't exist error, but I could see a situation where some other error gets returned. And if we decide after doing EFS testing that it's just noise, it would be simple to remove